bnm/recon/algo/service/volume.py

```diff
@@ -102,231 +106,217 @@ def vol_to_ext_surf_vol(self, in_vol_path, labels=None, hemi=None, out_vol_path=
+                    indexes_within_limits = numpy.all([(ijk_grid[:, 0] >= 0), (ijk_grid[:, 0] < mask_vol.dimensions[0]),
+                                                       (ijk_grid[:, 1] >= 0), (ijk_grid[:, 1] < mask_vol.dimensions[1]),
+                                                       (ijk_grid[:, 2] >= 0),
+                                                       (ijk_grid[:, 2] < mask_vol.dimensions[2])],
                                                       axis=0)
                     ijk_grid = ijk_grid[indexes_within_limits, :]
+
                     try:
                         # If none of these points is a mask point:
-                        if (mask[ijk_grid[:, 0], ijk_grid[:, 1], ijk_grid[:, 2]] < th).all():
-                            out_vol[i, j, k] = labels_nomask[iL]
+                        if (mask_vol.data[ijk_grid[:, 0], ijk_grid[:, 1], ijk_grid[:, 2]] < th).all():
+                            out_volume.data[current_voxel_i, current_voxel_j, current_voxel_k] = labels_nomask[label_index]
+
                         else:  # if any of them is a mask point:
-                            out_vol[i, j, k] = labels_mask[iL]
-                            out_ijk.append([i, j, k])
+                            out_volume.data[current_voxel_i, current_voxel_j, current_voxel_k] = labels_mask[label_index]
+                            out_ijk.append([current_voxel_i, current_voxel_j, current_voxel_k])
+
                     except ValueError:  # empty grid
-                        print "Error at voxel (" + str(i) + "," + str(j) + "," + str(k) + "):"
+                        print "Error at voxel (" + str(current_voxel_i) + "," + str(current_voxel_j) + "," + str(current_voxel_k) + "):"
                         print "It appears to have no common-face neighbors inside the image!"
                         return
+
                 else:
-                    out_vol[i, j, k] = labels_nomask[iL]
-        # Create the new volume and save it
-        out_volume = nibabel.Nifti1Image(out_vol, volume.affine, header=volume.header)
+                    out_volume.data[current_voxel_i, current_voxel_j, current_voxel_k] = labels_nomask[label_index]
+
         if out_vol_path == None:
-            # Overwrite volume
             out_vol_path = in_vol_path
-        nibabel.save(out_volume, out_vol_path)
-        # ...and the output indexes that survived masking
+
+        self.volume_io.write(out_vol_path, out_volume)
+
+        # Save the output indexes that survived masking
         out_ijk = numpy.vstack(out_ijk)
         filepath = os.path.splitext(out_vol_path)[0]
         numpy.save(filepath + "-idx.npy", out_ijk)
         numpy.savetxt(filepath + "-idx.txt", out_ijk, fmt='%d')
 
+    # Label one nifti with its dilation, cf seeg-ct.sh
     def label_with_dilation(self, to_label_nii_fname, dilated_nii_fname, out_nii_fname):
-        "Label one nifti with its dilation, cf seeg-ct.sh"
```

Comment:
A method comment is good to keep, just use this block after the method signature:
```
""""
Label one nifti with its dilation, cf seeg-ct.sh
""""
```